### PR TITLE
Added livenessprobe configuration example for CSI driver

### PR DESCRIPTION
**Pull request by sarjeet2013**
@msau42 @lpabon Please review.

I added the example configuration snippet from https://github.com/kubernetes-csi/livenessprobe#using-livenessprobe

### CODE REVIEW

File: book/src/Example.md [hunk 1]
```diff
@@ -133,6 +133,63 @@ csi-hostpathplugin-9rp7c     2/2     Running   0          5s
...
+      httpGet:
+        path: /healthz
+        port: healthz
+      initialDelaySeconds: 10
+      timeoutSeconds: 3
+      periodSeconds: 2
+      failureThreshold: 1
+...
+#
+# Spec for liveness probe sidecar container
```
Comment (msau42) [inline book/src/Example.md]: I would add some more detail here about what is going on.  This sidecar container is exposing the /healthz endpoint that the livenessProbe above queries.

File: book/src/Example.md [hunk 2]
```diff
@@ -133,6 +133,63 @@ csi-hostpathplugin-9rp7c     2/2     Running   0          5s
 {{#include example/hostpath/csi-hostpathplugin.yaml}}
 ```
 
+
+### Deploy livenessprobe with CSI plugin
+
+Below is an example configuration which needs to be added to CSI driver/provisioner yaml:
```
[the remark this answers is not in the document]
Comment (sarjeet2013) [reply book/src/Example.md]: provisioner I meant was controller. 

Will reword it to: "which needs to be added to CSI driver and/or controller yaml." 

Will this be fine?

File: book/src/Example.md (hunk 1 above)
Comment (sarjeet2013) [reply book/src/Example.md]: Sure, will add.

**Comment by sarjeet2013** [comment]
@msau42 Updated. PTAL

File: book/src/Example.md [hunk 3]
```diff
@@ -136,7 +136,11 @@ csi-hostpathplugin-9rp7c     2/2     Running   0          5s
 
 ### Deploy livenessprobe with CSI plugin
 
-Below is an example configuration which needs to be added to CSI driver/provisioner yaml:
+The CSI community provides a livenessprobe side-container that can be integrated with the CSI driver services (Node and Controller) to provide the liveness of the CSI service containers.
+
+The livenessprobe side-container will enable the integration with kubernetes liveness Probe, and will expose the endpoint on a httpGet request for CSI driver services. The same port is configured as `health-port` for liveness-probe side-container.
```
Comment (msau42) [inline book/src/Example.md]: Maybe reword to something like "The livenessprobe sidecar container will expose an http endpoint that be used in a Kubernetes liveness probe."

Comment (sarjeet2013) [reply book/src/Example.md]: done

**Comment by sarjeet2013** [comment]
@msau42 updated based on re-review comment.

File: book/src/Example.md [hunk 4]
```diff
@@ -133,6 +133,66 @@ csi-hostpathplugin-9rp7c     2/2     Running   0          5s
...
+    args:
+    - --csi-address=/csi/csi.sock
+    - --connection-timeout=3s
+    - --health-port=9898
+#
+```
+
+Where:
+
+- `--csi-address` - specifies the Unix domain socket path, on the host, for the CSI driver. It allows the livenessprobe sidecar to communicate with the driver for driver liveness information.  Mount path `/csi` is mapped to HostPath entry `socket-dir` which is mapped to directory `/var/lib/kubelet/plugins/csi-hostpath`
```
Comment (msau42) [inline book/src/Example.md]: so this is not actually the path on the host right? it's the path as seen in the container?

Comment (sarjeet2013) [reply book/src/Example.md]: @msau42  Yes. It will be path which will be seen inside container by livenessprobe side-container.

**Comment by msau42** [comment]
Thanks this lgtm, please squash all the commits

**Comment by sarjeet2013** [comment]
@msau42  Thanks for reviewing. I've squashed all commits to one. Please take a look and merge if looks good?

**Comment by saad-ali** [comment]
CC @sbezverk for FYI

/lgtm
